Common/TDSQASystemAPI/TestPackage/utils/TestPackageAssembler.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+using System.Xml.Serialization;
+
+namespace TDSQASystemAPI.TestPackage.utils
+{
+    /// <summary>
+    /// A class for building up a complete <code>TestPackage</code> after the XML has been deserialized.
+    /// </summary>
+    public class TestPackageAssembler
+    {

```

Comment:
This class was the simplest way I could come up with to make the parent -> child associations we need for getting certain values (e.g. the assessment key, the segment key, etc.).  Ideally, it'd be nice if the parent -> child relationship was set during the deserialization process.  I messed with the `XmlSerializer` quite a bit, but couldn't wrangle it to do what I wanted.  So here we are.